centos.sh

```diff
@@ -47,20 +46,20 @@ install_bison() {
     if [ "$OS_MAJOR" = "6" ]; then
         BISON_INSTALLED="false"
         if [ -x "$(command -v bison)" ]; then
-            BISON_VERSION=`bison --version | head -n 1 | awk '{print $4}'`
-            BISON_MAJOR=`echo $BISON_VERSION | cut -d. -f1`
+            BISON_VERSION=$(bison --version | head -n 1 | awk '{print $4}')
+            BISON_MAJOR=$(echo "$BISON_VERSION" | cut -d. -f1)
             if (( BISON_MAJOR >= 3 )); then
                 BISON_INSTALLED="true"
             fi
         fi
         if [ "$BISON_INSTALLED" = "false" ]; then
             wget https://ftp.gnu.org/gnu/bison/bison-3.0.4.tar.xz
             tar xvf bison-3.0.4.tar.xz
-            pushd bison-3.0.4
+            pushd bison-3.0.4 || return
```

Comment:
Elsewhere you used `exit 1`, which I think is better.  If we silently return, we are likely to get a cryptic error message later.

Reply:
Added different exit codes for errors within a file in all occurrences in 4650e8b54fcab4ef1c6a709a2cfdc21eadb2903b.